File: java/Sorting Algo/BubbleSort.java
public class BubbleSort 
{
    int[] bubbleSort(int arr[])
    {
        int n = arr.length;
        for (int i = 0; i < n - 1; i++)
        {
            for (int j = 0; j < n - i - 1; j++)
            {
                if (arr[j] > arr[j + 1]) 
                {
                    // swapping arr[j+1] and arr[j]
                    int temp = arr[j];
                    arr[j] = arr[j + 1];
                    arr[j + 1] = temp;
                }
            }
        }
        return arr;
    }
 
    void print(int arr[])
    {
        
        for (int i = 0; i < arr.length; i++)
            System.out.print(arr[i] + " ");
        System.out.println();
    }
 
    public static void main(String args[])
    {
        BubbleSort ob = new BubbleSort();
        int arr[] = { 9, 4, 2, 7, 5 };
        arr=ob.bubbleSort(arr);
        System.out.println("Sorted array:");
        ob.print(arr);
    }
}

File: C++ algos/Sorting Algos/bubbleSort.cpp
#include<bits/stdc++.h>
using namespace std ; 

// Function to Implement bubble Sort Algo
void bubble (int arr[] , int n){
    int i , j ;
    for(i = 0 ; i < n -1 ; i++){
        for(j = 0 ; j < n - i - 1; j++){
            if(arr[j] > arr[j + 1]){
                swap(arr[j], arr[j + 1]);
            }
        }
    }
}

// Function to print an array
void printArr(int arr[] , int n){
    for(int i = 0 ; i< n ; i++){
        cout<<arr[i]<<" ";
    }
    cout<<"\n";
}

int main() {
    int arr[] = {10, 5 , 6 , 4 , 8};
    int n = sizeof(arr) / sizeof(arr[0]);

    cout<<"Un-Sorted Array : "<<endl;
    printArr(arr,n);

    bubble(arr, n);

    cout<<"Sorted Array - "<<endl;
    printArr(arr,n);

    return 0;
}